### PR TITLE
[WIP] Added a11y section to the Button Readme.md

### PR DESCRIPTION
**Pull request by dpersing**
Currently marked as WIP since this PR has an issue where the section displays for Web, iOS, and Android in the style guide, although it should only be visible for Web. (See issue: https://github.com/Shopify/polaris-styleguide/issues/2421)

### WHY are these changes introduced?

Adds accessibility documentation and guidance for the Button component for consumption by the style guide.

Screenshot of button content displaying in the style guide

### WHAT is this pull request doing?

- Tweaked language around how buttons are activated to include keyboard users.
- Added best practices around using the accessibilityLabel prop and keyboard support.

### CODE REVIEW

File: src/components/Button/README.md [hunk 1]
```diff
@@ -100,6 +100,59 @@ Add a menu item
 
 <!-- end -->
 
+<!-- content-for: web -->
```
Comment (kaelig) [inline src/components/Button/README.md]: Let's take this off, as we'll extract the whole accessibility section out.

File: src/components/Button/README.md [hunk 2]
```diff
@@ -100,6 +100,59 @@ Add a menu item
...
+<button>Edit shipping address</button>
+```
+
+```
+<button aria-label="Edit shipping address">Edit</button>
+```
+
+#### Don’t
+
+Don't
```
[the remark this answers is not in the document]
Comment (dpersing) [reply src/components/Button/README.md]: Yes, since the Do/Don't icons don't have a text equivalent.

**Comment by dpersing** [comment]
@selenehinkley Right now this content would live in the main best practices and content guidelines section, if that's cool.

File: src/components/Button/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaelig) [reply src/components/Button/README.md]: Right, usageblock might be the thing you're looking for.

**Comment by dpersing** [comment]
Based on some more conversation with @selenehinkley, we're going to hold off on publishing for right now to help preserve the web/iOS/Android division. Let me know if you have questions!

**Pull request closed without merging**